### ISSUE TITLE
Remove testing dependencies from library dependencies

### ISSUE THREAD

**Issue by agrafix** [fpco/store#59]
Thanks for this awesome library! 

Are there any plans to remove `hspec` and friends from the library dependencies?

**Comment by Blaisorblade** [comment]
Yes please @mgsloan. Even installing `stack` requires `hspec` and related libs. And any client of `store` has to be built with the same version `hspec` as `store`, possibly requiring more recompilation for `cabal-install` users.

However, this is not so easy because of the use in Data.Store.TH, but that arguably belongs to a smaller `store-hspec` package. That's probably used in its own testsuite though, making the split a bit harder?
https://github.com/fpco/store/blob/ad2af60f1f65132c2a5b069f938ca25423a4b3e8/src/Data/Store/TH.hs

**Comment by mgsloan** [comment]
I think it makes sense to split off a `store-test` package, which does the following:

1) Defines utilities for conveniently defining tests for store instances
2) Has the store test-suite

**Comment by sol** [comment]
@mgsloan As a first step, wouldn't it make sense to just move the test dependencies from top-level to the test suite (haven't looked at it).

**Comment by sol** [comment]
> it also provides utilities for checking them via smallcheck and hspec

@mgsloan disregard what I said before, I just read your blog post. Makes sense now.

**Comment by mgsloan** [comment]
Cool!  Though, I am willing to revisit this if it helps make store more useful to people.

Could split things out into a `store-test` package.

**Comment by jrvieira** [comment]
Reviving this old but important issue!

Because store incorrectly declares hspec (and its transitive dependencies) as part of its library dependencies rather than confining it to a test suite, this forces every consumer of store to pull in hspec and its version‑constrained dependencies (e.g., QuickCheck) which conflict with other packages in the dependency graph leading to unnecessary and hard‑to‑debug solver failures even when the user never uses hspec themselves. I just spent an hour dealing with a solver deadlock due to this.

So I push to remove hspec, hspec-smallcheck, smallcheck, and other unnecessary dependencies from library dependencies.
